Change postgresql92 SCL references to rh-postgresql94.

jrafanie opened the pull request:
https://trello.com/c/aLn9eAj1

Related PRs:
https://github.com/ManageIQ/manageiq/pull/3676
https://github.com/ManageIQ/manageiq-appliance-build/pull/19

Depends on:
- [ ] Depends on https://github.com/ManageIQ/manageiq-appliance-build/pull/19

jrafanie committed:
Change postgresql92 SCL references to rh-postgresql94.

https://trello.com/c/aLn9eAj1

Related PRs:
https://github.com/ManageIQ/manageiq/pull/3676
https://github.com/ManageIQ/manageiq-appliance/pull/9 [this branch]
https://github.com/ManageIQ/manageiq-appliance-build/pull/19

jrafanie committed:
Add postgresql environment variables to DRY up ManageIQ code.

https://trello.com/c/aLn9eAj1

Related PRs:
https://github.com/ManageIQ/manageiq/pull/3676
https://github.com/ManageIQ/manageiq-appliance/pull/9 [this branch]
https://github.com/ManageIQ/manageiq-appliance-build/pull/19

jrafanie commented:
This is ready to be un-WIP'd and merged after ManageIQ/manageiq-appliance-build#19.

kbrock commented:
I think selinux will probably be using equivalence

can you ensure that the right selinux tag is on the data directory?

ls -Z /opt/rh/rh-postgresql94/root/var/lib/pgsql/data/pg_log/

jrafanie commented:
UGH, they changed the default data directory, < I think >

`ls -Z /opt/rh/rh-postgresql94/root/var/lib/pgsql/data/pg_log/` doesn't exist.  :cry: 

`/var/opt/rh/rh-postgresql94/lib/pgsql/data` looks to be the new location.  

So, choices are, set PGDATA environment variable and pass `-D #{PostgresAdmin.data_directory}` to `initdb` or use the new default location everywhere.

Thoughts @kbrock?

jrafanie commented:
In other words, it worked, we created the database disk in my test vm, but initdb setup the database in `/var/opt/rh/rh-postgresql94/lib/pgsql/data` and that's what used when the database starts.  It basically ignored our customization.

jrafanie commented:
cc @gtanzillo @abellotti Thoughts?

abellotti commented:
My take is leverage APPLIANCE_PG_DATA everywhere for the appliance, the only place you may not be able to use are the couple of places in the kickstart itself, which is ok for now.

jrafanie commented:
I went with the route of specifying the PGDATA env variable in the hopes that initdb and all postgres functions will use it to find the data directory.  The alternative is to be victim to the default data directory changing within the packaged postgresql.

Running a build with this to see how it goes.

kbrock commented:
We are going to use the same version of postgres for a while. Let not start adding all sorts of environment variables.

(Same as with Apache. We're not changing that directory name for the next year, adding an env variable is just adding complexity.)

jrafanie commented:
Yes @kbrock, it didn't work anyway, it didn't honor PGDATA environment variable.  Instead, I'll move all the hardcoded paths to the new `/var/opt/rh/rh-postgresql94/lib/pgsql/data` path.

jrafanie committed:
Change our database disk mount point to match changed defaults.

postresql92: /opt/rh/postgresql92/root/var/lib/pgsql/data
postresql94: /var/opt/rh/rh-postgresql94/lib/pgsql/data

https://trello.com/c/aLn9eAj1

Related PRs:
https://github.com/ManageIQ/manageiq/pull/3676
https://github.com/ManageIQ/manageiq-appliance/pull/9 [this branch]
https://github.com/ManageIQ/manageiq-appliance-build/pull/19

jrafanie committed:
Move PG files out of COPY so they're not copied at kickstart time.

PostgreSQL's data directory isn't initdb until first boot so any files
in the data directory will cause initdb to fail since it wants a completely
empty directory.

https://trello.com/c/aLn9eAj1

Related PRs:
https://github.com/ManageIQ/manageiq/pull/3676
https://github.com/ManageIQ/manageiq-appliance/pull/9 [this branch]
https://github.com/ManageIQ/manageiq-appliance-build/pull/19

jrafanie commented:
This is ready to go now.  I did a full end-to-end build with the other PRs and it works.  :sunglasses:

jrafanie commented on cfme-setup.sh:
```diff
@@ -32,7 +32,8 @@ EOF
 # relabel the pg_log directory in postgresql datadir, but defer restorecon
 # until after the database is initialized during firstboot configuration
-/usr/sbin/semanage fcontext -a -t var_log_t "/opt/rh/postgresql92/root/var/lib/pgsql/data/pg_log(/.*)?"
+/usr/sbin/semanage fcontext -a -t var_log_t "${APPLIANCE_PG_DATA}/pg_log(/.*)?"
```
@kbrock Can you confirm this looks ok?

```
[root@localhost ~]# ls -Z /var/opt/rh/rh-postgresql94/lib/pgsql/data/pg_log/
-rw-------. postgres postgres system_u:object_r:var_log_t:s0   postgresql.log
```

kbrock replied:
Looks like the old version, so this looks good to me. (thanks for checking)

```
 [root@localhost ~]# ls -lZ /opt/rh/postgresql92/root/var/lib/pgsql/data/pg_log/
-rw-------. postgres postgres system_u:object_r:var_log_t:s0   postgresql.log
```
